### ISSUE TITLE
CinemaZ does not use `PHPSESSID` (anymore?)

### ISSUE THREAD

**Issue by alvitali** [morpheus65535/bazarr#2797]
**Describe the bug**
Cannot add CinemaZ (and I assume AvistaZ, can not check though) as a subitle provider in Bazarr, as the `PHPSESSID` token is not displayed for CinemaZ anymore using dev tools.

**To Reproduce**
1. Log in to CinemaZ
2. Open Dev Tools > Storage > Cookies > https://cinemaz.to
3. Reload the page
4. No `PHPSESSID` token available, only `cinemazx_session`

**Expected behavior**
Possibility to add CinemaZ to subtitle providers again, or updated information on how to do that

**Screenshots**

**Software (please complete the following information):**
 - Bazarr: v1.4.5
 - Radarr version: not relevant to issue
 - Sonarr version: not relevant to issue
 - OS: Docker, Debian 12

**Additional context**

**Comment by morpheus65535** [comment]
@goroskob would you take a look at this issue please?

**Comment by goroskob** [comment]
PHPSESSID cookie was in fact never a requirement for the provider to work. The cookies string that have been working for me for months doesn't have it. The text over the input field ('Cookies, e.g., PHPSESSID=abc; wikisubtitlesuser=xyz; wikisubtitlespass=efg') was always intended just an example of how cookies string should look like.

I didn't see a point in investigating what exact cookies must be used for the provider to work, but the ones I have working are: XSRF-TOKEN, cinemazx_session, cf_clearance, love, remember_web_*.

**Comment by morpheus65535** [comment]
@goroskob sorry for the delay to answer.

This provider had about 20 downloads from all users in the last 3 months... does it really works or is it just not used that much?

**Comment by goroskob** [comment]
@morpheus65535 that’s quite strange. My instance of Bazarr alone grabbed more than a hundred subtitles from AvistaZ in just the last month. I have the Analytics enabled in the settings, but I guess the data doesn’t make it to you. 

Unfortunately, I have no idea how many more people have the provider set up and working, as we kinda have a reverse survivorship bias with people only sharing when something goes wrong.

**Comment by morpheus65535** [comment]
@goroskob I've checked only for Cinemaz, Avistaz got 521 downloads in the last month.

**Comment by goroskob** [comment]
@morpheus65535 understood. I will take a look if anything has broken on CinemaZ provider when I have the time, but probably just not that many users. AvistaZ has 4 times the user base of CinemaZ, and very different content which may or may not trigger less downloads due to built-in subtitles in video files.

**Comment by Jacco011** [comment]
CinemaZ and AvistaZ use three different cookies. It seems to me that cf_clearance must be the right one.
- XSRF-TOKEN (under Response Headers) but these expires fast Max-Age=2628000.
- cinemazx_session (under Response Headers) Max-Age=2628000.
- avistazx_session (under Response Headers Max-Age=216000.
- cf_clearance (Request Headers) no max-age.

**Comment by stealthytanker** [comment]
> CinemaZ and AvistaZ use three different cookies. It seems to me that cf_clearance must be the right one.
> 
> * XSRF-TOKEN (under Response Headers) but these expires fast Max-Age=2628000.
> * cinemazx_session (under Response Headers) Max-Age=2628000.
> * avistazx_session (under Response Headers Max-Age=216000.
> * cf_clearance (Request Headers) no max-age.

Do cf_clearance and it’s related cookie tied to an ip? My avistaz cookie session keeps expiring from time to time for bazarr use

**Comment by hlibkoval** [comment]
> Do cf_clearance and it’s related cookie tied to an ip? My avistaz cookie session keeps expiring from time to time for bazarr use

Interesting. My set of cookies has been valid for at least a year now, maybe more. Static IP addresses though

**Comment by stealthytanker** [comment]
> > Do cf_clearance and it’s related cookie tied to an ip? My avistaz cookie session keeps expiring from time to time for bazarr use
> 
> Interesting. My set of cookies has been valid for at least a year now, maybe more. Static IP addresses though

I guess long term cookie can be reached only if you get cookie from static IP seed box or your own bazarr server at home. CMIIW

**Comment by joedalton17** [comment]
Does this provider still work and does it get any downloads?
Because it tried to add and test every version of cookies i could finde for both CinemaZ and AvistaZ and none worked. So i am really wondering if other people get hits or not?
I checked my logs and have had no downloads for the last 6 months. 
The error in bazarr is AuthenticationError.

**Comment by hlibkoval** [comment]
Hey @joedalton17 . Just went and checked my instance - I did have AuthenticationError on the Avistaz provider too (idk for how long, that set of cookies worked for me for at least a year), but with a clear "'cookies not valid anymore" message in the logs. I logged in in Chrome and got the whole "Cookie" header from the Request Headers in devtools, updated the provider settings with them. I was able to successfully grab subtitles from Avistaz after that.

UPDATE tested Cinemaz provider too. Works for me with some old cookies (haven't updated them in ages).

**Comment by joedalton17** [comment]
Hello @hlibkoval thanks for looking into it. I have only tried it with Firefox and it did not work. Now i tried with Chrome and it worked so i might be a Firefox thing.
For other people:
F12 in Chrome --> Head to the Network tab
Press F5 and on the left side there should be pictures, svg and other things. One should have the website name -> klick it
Klick on "Header" and scroll down to "Request Headers" and then copy the whole "Cookie" tab to Bazaar
After you are done scroll all the way down to grab the "User-Agent"

**Comment by Asinin3** [comment]
@joedalton17 I can confirm same issue with Firefox, i cleared my browser data in firefox and logged back in again, but did not get all the cookies needed for bazarr to work with it. It didn't pull these cookies: _ga=, _gid=, XSRF-TOKEN, avistazx_session .. and some remember web one.

But it turns out you can just click the raw data button in firefox to show the full cookie:
<img width="1885" height="363" alt="Image" src="https://github.com/user-attachments/assets/dad056e6-22d9-43f8-bfe1-7b67db3673f6" />

**Comment by luuvasara** [comment]
**TL;DR:** @hlibkoval already nailed this above — `PHPSESSID` was never required; the field's example text was just misleading. So this isn't a provider bug, it's a help-text bug where **AvistaZ** and **CinemaZ** cookie field points people at cookies that don't exist.

What you actually need is the **full `Cookie` header from a logged-in browser** (the session cookie + `cf_clearance`) and a **matching User-Agent** — exactly what @joedalton17 and @Asinin3 described.

<details>
<summary><b>Why — confirmed against the code</b></summary>

In `avistaz_network.py` → `initialize()`:

- It loads **whatever you paste**: `SimpleCookie().load(self.cookies)`, then sets every cookie generically — there is **no `PHPSESSID`** anywhere in the provider or `get_providers.py`. (Confirms @hlibkoval.)
- Auth check: `GET {server_url}rules` with no redirects; `302/404/403` → `raise AuthenticationError("cookies not valid anymore")` — that's the error people hit when the cookies are wrong/expired.
- The session is a `RetryingCFSession()` (Cloudflare), so `cf_clearance` + a matching User-Agent matter — and because `cf_clearance` is UA/IP-bound, that's why @stealthytanker's cookie expired on an IP change.

So the only thing actually broken was the example text:
`Cookies, e.g., PHPSESSID=abc; wikisubtitlesuser=xyz; wikisubtitlespass=efg`
— `PHPSESSID` is gone, and `wikisubtitlesuser/pass` belong to a different provider entirely.
</details>

<details>
<summary><b>Working setup (from @joedalton17 / @Asinin3)</b></summary>

1. Log in to `cinemaz.to` / `avistaz.to` in your browser.
2. DevTools → **Network** tab → reload the page → click any request to the site.
3. **Request Headers → `Cookie`** → copy the whole value.
   *(Firefox: enable the **Raw** toggle, otherwise some cookies are hidden.)*
4. Paste it into Bazarr's **Cookies** field, and copy the **User-Agent** into the User-Agent field.
</details>

**Fix (help-text only):** update the AvistaZ + CinemaZ cookie field labels

*(The same stale example also appears on `addic7ed` and `turkcealtyaziorg`;

**Comment by morpheus65535** [comment]
Fixed the help-text in upcoming beta. I'll close this issue. Let me know if there's something else to fix.
